apps/dashboard/src/components/primitives/form/faceted-filter/components/base-filter-content.tsx

```diff
@@ -0,0 +1,74 @@
+import { SizeType } from '../types';
+import { FilterInput } from './filter-input';
+import { ClearButton } from './clear-button';
+import { RiArrowDownLine, RiArrowUpLine } from 'react-icons/ri';
+import { EnterLineIcon } from '../../../../icons/enter-line';
+
+interface BaseFilterContentProps {
+  inputRef: React.RefObject<HTMLInputElement>;
+  title?: string;
+  onClear: () => void;
+  size: SizeType;
+  hideSearch?: boolean;
+  hideClear?: boolean;
+  searchValue?: string;
+  onSearchChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  searchPlaceholder?: string;
+  showNavigationFooter?: boolean;
+  showEnterIcon?: boolean;
+  children?: React.ReactNode;
+}
+
+export function BaseFilterContent({
+  inputRef,
+  title,
+  onClear,
+  size,
+  hideSearch = false,
+  hideClear = false,
+  searchValue = '',
+  onSearchChange,
+  searchPlaceholder,
+  showNavigationFooter = false,
+  showEnterIcon = false,
+  children,
+}: BaseFilterContentProps) {
+  return (
+    <div className="p-0">
+      <div className="flex justify-between rounded-t-md bg-neutral-50 px-1.5 py-1">
+        {title && <span className="text-foreground-400 text-[11px] uppercase">{title}</span>}
+        {!hideClear && searchValue && <ClearButton onClick={onClear} size={size} label="Reset" />}
+      </div>
+
+      {!hideSearch && onSearchChange && (
+        <FilterInput
+          inputRef={inputRef}
+          value={searchValue}
+          onChange={onSearchChange}
+          placeholder={searchPlaceholder}
+          size={size}
+          showEnterIcon={showEnterIcon}
+        />
+      )}
+
+      {children}
+
+      {showNavigationFooter && (
+        <div className="flex justify-between rounded-b-md border-t border-neutral-100 bg-white p-1">
+          <div className="flex items-center gap-0.5">
+            <div className="pointer-events-none shrink-0 rounded-[6px] border border-neutral-200 bg-white p-1 shadow-[0px_0px_0px_1px_rgba(14,18,27,0.02)_inset,_0px_1px_4px_0px_rgba(14,18,27,0.12)]">
```

Comment:
Our shadows currently not synchronized with Figma. Will schedule a session with naveen to fix it across the board. Added it manually here for now